### PR TITLE
Extend runtime exectution page

### PR DESCRIPTION
**Pull request by bulislaw**
@sg- @AnotherButler

### CODE REVIEW

File: docs/reference/runtime/Execution.md [hunk 1]
```diff
@@ -1,6 +1,25 @@
 ## Execution
 
-You can use the same RTOS API in ISR. The only two warnings are:
+### Threads
+
+Your application (`main` function) will start execution in the main thread, but it's not the only thread running in Mbed OS. There's a number of live system threads at any given time. They should be transparent from the user's point of view for most use cases. All threads that exist in the system by default:
+* main - default thread that executes application's `main` function.
```
Comment (sg-) [inline docs/reference/runtime/Execution.md]: main thread - ... int main void is the main system thread and has stack size of ... by default

File: docs/reference/runtime/Execution.md [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+* idle - thread that's run by the scheduler when there's no other activity in the system (e.g. all other threads are waiting for some event). It's used to make sure the board is not burning empty processor cycles, but is put to sleep for as long as possible.
```
Comment (sg-) [inline docs/reference/runtime/Execution.md]: idle thread - comment about centralized sleep management?

File: docs/reference/runtime/Execution.md [hunk 3]
```diff
@@ -1,6 +1,25 @@
...
+* timer - thread for handling system and user timer objects.
```
Comment (sg-) [inline docs/reference/runtime/Execution.md]: Is this really still used? Can we say its depreciated with the RtosTimer class?

Comment (bulislaw) [reply docs/reference/runtime/Execution.md]: Good point will add that.

File: docs/reference/runtime/Execution.md (hunk 2 above)
Comment (bulislaw) [reply docs/reference/runtime/Execution.md]: By centralised sleep management you mean?

File: docs/reference/runtime/Execution.md (hunk 1 above)
Comment (bulislaw) [reply docs/reference/runtime/Execution.md]: I've added that as a note after, but it fits here better, I'll move that info inline.

**Comment by AnotherButler** [comment]
@bulislaw Please review my changes, and make sure I did not accidentally change the meaning of the content. Please pay special attention to the `main` bullet. I tried to clarify but may have misunderstood.

**Comment by bulislaw** [comment]
@AnotherButler thanks for that, looks good!

**Pull request merged**